src/XMakeBuildEngine/Definition/ToolsetReader.cs

```diff
@@ -204,7 +204,7 @@ ToolsetDefinitionLocations locations
 
                         // Other toolsets are installed in the xbuild directory
                         var xbuildToolsetsDir = Path.Combine(libraryPath, "xbuild");
-                        var r = new Regex(xbuildToolsetsDir + Path.DirectorySeparatorChar + @"\d+\.\d+");
+                        var r = new Regex(Regex.Escape(xbuildToolsetsDir + Path.DirectorySeparatorChar) + @"\d+\.\d+");
```

Comment:
And you could move that `var r = new Regex...` inside the directory.exists block.